examples/step-97/step-97.cc

```diff
@@ -0,0 +1,2125 @@
+  // \f]
+  void Solver::solve()
+  {
+    SolverControl control(system_rhs.size(),
+                          1.0e-6 * system_rhs.l2_norm(),
+                          false,
+                          false);
+
+    if (Settings::log_cg_convergence)
+      control.enable_history_data();
+
+    GrowingVectorMemory<Vector<double>> memory;
+    SolverCG<Vector<double>>            cg(control, memory);
+
+    PreconditionSSOR<SparseMatrix<double>> preconditioner;
+    preconditioner.initialize(system_matrix, 1.2);
+
+    cg.solve(system_matrix, solution_Hdiv, system_rhs, preconditioner);
+
+    if (Settings::log_cg_convergence)
+      {
+        const std::vector<double> history_data = control.get_history_data();
+
+        std::ofstream ofs(fname + "_cg_convergence.csv");
+
+        unsigned int i = 1;
+        for (auto item : history_data)
+          {
+            ofs << i << ", " << item << "\n";
+            i++;
+          }
+        ofs.close();
+      }
+  }
+
+  // This function saves the computed fields into a vtu file. This time we also
+  // save the projected exact solution and the $L^2$ error norm. The exact
+  // solution is only saved if the `Settings::project_exact_solution = true`
+  void Solver::save() const
+  {
+    std::vector<std::string> solution_names(3, "VectorField");
+    std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      interpretation(3,
+                     DataComponentInterpretation::component_is_part_of_vector);
+
+    DataOut<3> data_out;
+
+    data_out.add_data_vector(dof_handler_Hdiv,
+                             solution_Hdiv,
+                             solution_names,
+                             interpretation);
+
+    if (Settings::project_exact_solution)
+      {
+        std::vector<std::string> solution_names_ex(3, "VectorFieldExact");
+
+        data_out.add_data_vector(dof_handler_Hdiv,
+                                 projected_exact_solution,
+                                 solution_names_ex,
+                                 interpretation);
+      }
+
+    if (exact_solution)
+      {
+        data_out.add_data_vector(L2_per_cell, "L2norm");
+      }
+
+    std::ofstream ofs;
+
+    DataOutBase::VtkFlags flags;
+    flags.write_higher_order_cells = true;
+    data_out.set_flags(flags);
+
+    const MappingQ<3> mapping(mapping_degree);
+
+    data_out.build_patches(mapping,
+                           fe_Hdiv.degree + 2,
+                           DataOut<3>::CurvedCellRegion::curved_inner_cells);
+
+    ofs.open(fname + ".vtu");
+    data_out.write_vtu(ofs);
+
+    ofs.close();
+  }
+} // namespace ProjectorHcurlToHdiv
+
+
+// @sect3{The main loop}
+
+// This class contains the main loop of the program.
+class Batch
+{
+public:
+  void run()
+  {
+    if (Settings::nr_threads_max)
+      MultithreadInfo::set_thread_limit(Settings::nr_threads_max);
+
+    const unsigned int m = 2; // Mapping degree.
+    const unsigned int p = Settings::fe_degree;
```

Comment:
Since you already have `Settings::fe_degree`, would it be useful to also add `Settings::mapping_degree`?